### ISSUE TITLE
Configuring Akka to use Serilog logger causes high CPU and memory usage

### ISSUE THREAD

**Issue by object** [akkadotnet/akka.net#1724]
Our test application (using Akka.NET 1.0.6 and F# API) continuously use 30-40% CPU and its memory consumption increases all the time, so it eventually crashes with OutOfMemory exception. This happens even if I comment out almost all application functionality leaving just a line where it creates Akka system. It is still a CPU and memory hog. Selectively disabling the bits of the application I noticed that all CPU/memory problems disappear once I remove the following line from Akka HOCON configuration:

            loggers = ["Akka.Logger.Serilog.SerilogLogger, Akka.Logger.Serilog"]                                

What is interesting is that CPU consumption is still low if continue using Serilog by calling its logger directly. It logs without high CPU or memory usage. But if bring back the line above, the problems reappear.

Another interesting thing is that even when I configure Akka to use Serilog, I don't see any Akka system messages in the log, only those my app writes directly. So obviously system logging doesn't work.

The problem occurs even if Serilog configuraiton is the only line in Akka HOCON string:

    <hocon>
      <![CDATA[
          akka {                
            loggers = ["Akka.Logger.Serilog.SerilogLogger, Akka.Logger.Serilog"]                                
          }
      ]]>
    </hocon>

**Comment by tstojecki** [comment]
> Another interesting thing is that even when I configure Akka to use Serilog, I don't see any Akka system messages in the log, only those my app writes directly.

Do you have a log level set to DEBUG? Most of the system level messages are DEBUG level and without specifying in HOCON, it will default to INFO.

**Comment by rogeralsing** [comment]
The system messages are not semantic, we log our internal messages using the old string format convention `Foo {0} Bar {1}` which is incompatible with Serilog which expects named placeholders.

My guess is that there is some weird parsing or transformation of log messages going on here.

**Comment by annymsMthd** [comment]
I'm having a problem reproducing this issue. Maybe I'm missing something with the Serilog config?

**Comment by object** [comment]
I will try to extract a small example that reproduces the issue. But most of config is like this:


    let configureLogging (applicationName : string) =
        LogContext.PermitCrossAppDomainCalls <- true

        let logRoot = "CHOOSE_YOUR_LOG_ROOT" 
        let logSink = new RollingFileSink(
                        Path.Combine(logRoot, String.Format("{0}.{{Date}}.txt", applicationName)),
                        JsonFormatter(true), Nullable(), Nullable(), UTF8Encoding(false))

        let config = LoggerConfiguration().WriteTo.ColoredConsole().MinimumLevel.Debug().WriteTo.Sink(logSink)
        let logger = config.Enrich.FromLogContext().MinimumLevel.Debug().CreateLogger()
        Serilog.Log.Logger <- logger

**Comment by object** [comment]
I managed to reproduce the problem in a tiny app. Here's the repo: https://github.com/object/AkkaSerilogTest

On our development machines once I start that console app, CPU usage stays on 30-40% and memory consumption quickly goes up until the program runs out of memory.

**Comment by Danthar** [comment]
**TL;DR: Its fixed in 1.0.7**

Ok some observations:
This only happens in **v1.0.6**
The problem occurs when you turn on the **log all the things!** equivalent for Akka

```
 actor {
              debug {
                receive = on # log any received message
                autoreceive= on # log automatically received messages, e.g. PoisonPill
                lifecycle = on # log actor lifecycle changes
                fsm = off # log all LoggingFSMs for events, transitions and timers
                event-stream = on # log subscription changes for Akka.NET event stream
                unhandled = on # log unhandled messages sent to actors
              }
            }
``` 
Internally this causes an FileLoadException to be thrown on a loop. I surmised this has something to do with the initialisation of the Logging infrastructure in combination with some timing related problem.

Because in this configuration we log everything, we also attempt to log the Exception. Which causes an exception which is logged which..... etc et infinitum. 

This is the cause of your Memory leak.

This problem does **not** occur in **v1.0.7**
As to the why its solved. I haven't looked into that to deeply. Comparing changes between 1.0.6 and 1.0.7 of the relevant namespaces reveal multiple possible sources. The biggest contenders are probably:
https://github.com/akkadotnet/akka.net/commit/18ef093f8fd33fdee9b6c3888f18026df5914ee4#diff-8a3a177d755ec44f8dde7e7c60282f9e
and 
https://github.com/akkadotnet/akka.net/commit/af36e1a606d4d8f9197baf2d6c39bd63b0f993d2#diff-8a3a177d755ec44f8dde7e7c60282f9e
